hummingbot/connector/exchange/hyperliquid/hyperliquid_api_order_book_data_source.py

```diff
@@ -0,0 +1,149 @@
+import asyncio
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+# from bidict import bidict
+from hummingbot.connector.exchange.hyperliquid import (
+    hyperliquid_constants as CONSTANTS,
+    hyperliquid_web_utils as web_utils,
+)
+from hummingbot.connector.exchange.hyperliquid.hyperliquid_order_book import HyperliquidOrderBook
+from hummingbot.core.data_type.order_book_message import OrderBookMessage
+from hummingbot.core.data_type.order_book_tracker_data_source import OrderBookTrackerDataSource
+from hummingbot.core.web_assistant.connections.data_types import WSJSONRequest
+from hummingbot.core.web_assistant.web_assistants_factory import WebAssistantsFactory
+from hummingbot.core.web_assistant.ws_assistant import WSAssistant
+from hummingbot.logger import HummingbotLogger
+
+if TYPE_CHECKING:
+    from hummingbot.connector.exchange.hyperliquid.hyperliquid_exchange import HyperliquidExchange
+
+
+class HyperliquidAPIOrderBookDataSource(OrderBookTrackerDataSource):
+    HEARTBEAT_TIME_INTERVAL = 30.0
+    TRADE_STREAM_ID = 1
+    DIFF_STREAM_ID = 2
+    ONE_HOUR = 60 * 60
+
+    _logger: Optional[HummingbotLogger] = None
+
+    def __init__(self,
+                 trading_pairs: List[str],
+                 connector: 'HyperliquidExchange',
+                 api_factory: WebAssistantsFactory,
+                 domain: str = CONSTANTS.DOMAIN):
+        super().__init__(trading_pairs)
+        self._connector = connector
+        self._trade_messages_queue_key = CONSTANTS.TRADE_EVENT_TYPE
+        self._diff_messages_queue_key = CONSTANTS.DIFF_EVENT_TYPE
+        self._domain = domain
+        self._api_factory = api_factory
+
+    async def get_last_traded_prices(self,
+                                     trading_pairs: List[str],
+                                     domain: Optional[str] = None) -> Dict[str, float]:
+        return await self._connector.get_last_traded_prices(trading_pairs=trading_pairs)
+
+    async def _request_order_book_snapshot(self, trading_pair: str) -> Dict[str, Any]:
+        ex_trading_pair = await self._connector.exchange_symbol_associated_to_pair(trading_pair=trading_pair)
+        coin = ex_trading_pair.replace("-", "/")
```

Comment:
this should be already solved when getting the exchange trading pair from the map. are you getting the trading rules correctly?

Reply:
oh true since my trading rules works, it should work without the self._connector.name_to_coin[coin] conversion,  One moment @cardosofede

Reply:
works fine without the conversion, trading rules sorted that already, will refactor and push